packages/itwin/reports-config-widget/src/widget/hooks/useValidator.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelApp } from "@itwin/core-frontend";
+import { useState } from "react";
+import SimpleReactValidator from "simple-react-validator";
+
+export const NAME_REQUIREMENTS =
+  "required|NoDuplicateUnderscore|OnlyBeginsWithLetterOrUnderscore|FollowedByLettersUnderscoresSpacesAndDigits|CharLimit";
+
+const useValidator = (
+): [SimpleReactValidator, React.Dispatch<React.SetStateAction<boolean>>] => {
+  const [show, setShow] = useState(false);
+  const customValidator = {
+    NoDuplicateUnderscore: {
+      message: IModelApp.localization.getLocalizedString("ReportsConfigWidget:Validators.NoDuplicateUnderscore"),
```

Comment:
instead of IModelApp.localization shouldnt you just use ReportsConfigWidgget.i18n?

Reply:
True. Should have switched it over when I made the class.